pyuvdata/tests/test_telescopes.py

```diff
@@ -124,44 +143,183 @@ def test_known_telescopes():
+
+    with uvtest.check_warnings(
+        DeprecationWarning,
+        match="The Telescope.telescope_location attribute is deprecated, use "
+        "Telescope.location instead (which contains an astropy "
+        "EarthLocation object). This will become an error in version 3.2.",
+    ):
+        np.testing.assert_allclose(mwa_tel.telescope_location, mwa_tel._location.xyz())
+
+    with uvtest.check_warnings(
+        DeprecationWarning,
+        match="The Telescope.telescope_name attribute is deprecated, use "
+        "Telescope.name instead. This will become an error in version 3.2.",
+    ):
+        mwa_tel.telescope_name = "foo"
+    assert mwa_tel.name == "foo"
+
+    hera_tel = Telescope.from_known_telescopes("hera")
+    with uvtest.check_warnings(
+        DeprecationWarning,
+        match="The Telescope.telescope_location attribute is deprecated, use "
+        "Telescope.location instead (which should be set to an astropy "
+        "EarthLocation object). This will become an error in version 3.2.",
+    ):
+        mwa_tel.telescope_location = hera_tel._location.xyz()
+    assert mwa_tel._location == hera_tel._location
+
+
+@pytest.mark.filterwarnings("ignore:Directly accessing the KNOWN_TELESCOPES")
+def test_old_known_tel_dict_keys():
+    from pyuvdata.telescopes import KNOWN_TELESCOPES
+
+    hera_tel = Telescope.from_known_telescopes("hera")
+
+    warn_msg = [
+        "Directly accessing the KNOWN_TELESCOPES dict is deprecated. If you "
+        "need a telescope location, use the known_telescope_location function. "
+        "For a full Telescope object use the classmethod "
+        "Telescope.from_known_telescopes."
+    ] * 2
+
+    with uvtest.check_warnings(DeprecationWarning, match=warn_msg):
+        assert KNOWN_TELESCOPES["HERA"]["latitude"] == hera_tel.location.lat.rad
+
+    with uvtest.check_warnings(DeprecationWarning, match=warn_msg):
+        assert KNOWN_TELESCOPES["HERA"]["longitude"] == hera_tel.location.lon.rad
+
+    with uvtest.check_warnings(DeprecationWarning, match=warn_msg):
+        assert (
+            KNOWN_TELESCOPES["HERA"]["altitude"]
+            == hera_tel.location.height.to("m").value
+        )
+
+    with uvtest.check_warnings(DeprecationWarning, match=warn_msg):
+        np.testing.assert_allclose(
+            KNOWN_TELESCOPES["HERA"]["center_xyz"],
+            Quantity(hera_tel.location.geocentric).to("m").value,
+        )
+    with uvtest.check_warnings(DeprecationWarning, match=warn_msg):
+        assert KNOWN_TELESCOPES["HERA"]["citation"] == hera_tel.citation
+
+    assert len(KNOWN_TELESCOPES["MWA"]) == 1
+    for key, val in KNOWN_TELESCOPES.items():
+        assert val == _KNOWN_TELESCOPES[key]
 
 
 def test_get_telescope_center_xyz():
     ref_xyz = (-2562123.42683, 5094215.40141, -2848728.58869)
-    ref_latlonalt = (-26.7 * np.pi / 180.0, 116.7 * np.pi / 180.0, 377.8)
+    # ref_latlonalt = (-26.7 * np.pi / 180.0, 116.7 * np.pi / 180.0, 377.8)
```

Comment:
Cruft?

Reply:
yes, thanks!

Reply:
Actually, I'm not sure that test is needed. I'll do a little checking on that, I might remove the whole test.

Reply:
I removed it entirely.